plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/wizard/dependencies/RunnableInstallDependnecies.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2018 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ui.wizard.dependencies;
+
+import java.io.File;
+import java.net.URI;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.MultiStatus;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.jface.operation.IRunnableWithProgress;
+import org.eclipse.n4js.binaries.BinariesPreferenceStore;
+import org.eclipse.n4js.binaries.nodejs.NpmrcBinary;
+import org.eclipse.n4js.external.NpmLogger;
+import org.eclipse.n4js.ui.utils.AutobuildUtils;
+import org.eclipse.n4js.utils.StatusHelper;
+
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+/**
+ * Runnable collector of the workspace projects setting files. Note single instance single use, not thread safe.
+ */
+public class RunnableInstallDependnecies implements IRunnableWithProgress {
+	private MultiStatus multistatus;
+	private InstallOptions options;
+
+	@Inject
+	private StatusHelper statusHelper;
+
+	@Inject
+	private ExternalLibrariesInstallHelper librariesInstallHelper;
+
+	@Inject
+	private BinariesPreferenceStore preferenceStore;
+
+	@Inject
+	private Provider<NpmrcBinary> npmrcBinaryProvider;
+
+	/** Logger that logs to the npm console which is visible by the user in his IDE. */
+	@Inject
+	private NpmLogger userLogger;
+
+	/** needs to be called before {@link #run(IProgressMonitor)} */
+	public void setInstallOptions(InstallOptions options) {
+		this.options = options;
+	}
+
+	/** @return status of runnable */
+	public synchronized IStatus getResultStatus() {
+		if (multistatus == null)
+			return statusHelper.createError(getClass().getName() + " was not called yet!");
+
+		MultiStatus result = statusHelper.createMultiStatus("result");
+		result.merge(multistatus);
+		return result;
+	}
+
+	@Override
+	synchronized public void run(IProgressMonitor pmonitor) {
+		final SubMonitor monitor = SubMonitor.convert(pmonitor, 100);
+		final SubMonitor subMonitor2 = monitor.split(1);
+
+		final boolean wasAutoBuilding = AutobuildUtils.get();
+
+		AutobuildUtils.turnOff();
```

Comment:
you can use the Autoclosable to temporarily turn off autobuild

Reply:
but I want to close it *only* if there will be no exceptions... so no?

Reply:
k